hardware_interface/include/hardware_interface/utils/component_parser.hpp

```diff
@@ -52,7 +53,10 @@ ComponentInfo parse_component_from_xml(const tinyxml2::XMLElement * component_it
  * \return std::map< std::__cxx11::string, std::__cxx11::string > key-value map with parameters
  * \throws std::runtime_error if a component attribute or tag is not found
  */
-ROBOT_CONTROL_COMPONENTS_PUBLIC
-std::map<std::string, std::string> parse_parameters_from_xml(const tinyxml2::XMLElement * params_it);
-}  // namespace robot_control_components
-#endif  // ROBOT_CONTROL_COMPONENTS__COMPONENT_PARSER_HPP_
+HARDWARE_INTERFACE_PUBLIC
+std::map<std::string, std::string> parse_parameters_from_xml(
```

Comment:
does that have to be a sorted map though? Would `std::unordered_map` work as well?

Reply:
You are right!

Reply:
solved in ros-controls/ros2_control#101 84c873c